CRM/Utils/System/Base.php

```diff
@@ -1051,6 +1051,23 @@ public function getCMSPermissionsUrlParams() {
     return [];
   }
 
+  /**
+   * Get the CMS database as a 'prefix'.
+   *
+   * This returns a string that can be prepended to a query to include a CMS table.
+   *
+   * However, this string should contain backticks, or not, in accordance with the
+   * CMS's drupal views expectations, if any.
+   */
+  public function getCMSDatabasePrefix(): string {
+    $crmDatabase = DB::parseDSN(CRM_Core_Config::singleton()->dsn)['database'];
+    $cmsDatabase = DB::parseDSN(CRM_Core_Config::singleton()->userFrameworkDSN)['database'];
+    if ($crmDatabase === $cmsDatabase) {
+      return '';
+    }
+    return "`$cmsDatabase`.";
```

Comment:
I was just testing this (drupal 7) - this is backwards - you need the crmdatabase.

Reply:
@demeritcowboy do you want to push in the changes that this PR needs?